website/docs/pact_nirvana/step_7.md

```diff
@@ -22,36 +31,78 @@ Another reason that an extra "safe to deploy?" check is required (as well as the
-  * You may wish to start off using the `--dry-run` flag for `can-i-deploy`, which will allow you to see the results, but won't fail your build if there are missing or failed verifications. Once you are confident that your pipeline is working as desired, you can remove the flag, and be confident that you won't be allowed to deploy any incompatible changes to an environment.
-3. Add a step to your deployment process so that when the application is deployed to production, the relevant pacticipant version in the broker is tagged as the “production” version. This functionality is provided by the [pact-broker client CLI](/pact_broker/client_cli/readme#create-version-tag)
 
-Useful links:
+## Lets tag on deploy!
 
-* [Checking verification results using the can-i-deploy CLI](/pact_broker/can_i_deploy/)
-* [can-i-deploy CLI documentation](/pact_broker/client_cli/readme#can-i-deploy)
+Now we're ready to tag pacts when we deploy, and switch `can-i-deploy` to use environment names rather than branch names.
+
+<details open>
+  <summary>Diamond level diagram</summary>
+
+![Diamond level diagram](images/diamond.png)
+</details>
+
+Note how in the PR pipeline we are checking to see if we can deploy to all the environments that we deploy to from 
+the main branch. The rationale for this is that we don't want to merge something into our main branch which we know 
+will break once we deploy - we want our main branch to stay clean and not get blocked by broken builds.  If you 
+deploy to other environments such as preprod, you would want to check that environment as well.
+
+Then, when we're actually ready to deploy to an environment, we check again. We do this because the version of the 
+other side of the contract could have changed since the PR validation ran.  Doing the check in the PR pipeline 
+reduces the chance of breakage, but it's still a possibility, and we want to check for that before we deploy.
+
+Note also that the first step in our commit pipeline is to let the broker know that a particular version of the 
+provider or consumer is now in the main branch.
+
+### Add tagging to your commit pipelines
 
+The first step is to start tagging when the consumer or provider is deployed to an environment. So when you successfully 
+deploy to staging, apply the staging tag; when you successfully deploy to prod, apply the prod tag. Do this for each 
+environment you deploy to. 
 
-## Setting up a Pact in a "universal" CI/CD pipeline
+1. Add a step to your deployment process so that when the consumer is deployed to an environment, the relevant pacticipant version in the broker is tagged with the name of the environment (eg. `test`, `production`). This functionality is provided by the [pact broker CLI](/pact_broker/client_cli/readme#create-version-tag) which is available as a standalone executable, and a Docker image.
```

Comment:
The CLI is a bit tricky to work with. We have ended up using the REST API directly. Perhaps you can mention that and provide a link to the API documentation?

Reply:
Certainly, I've just merged a PR which fixes some broken links, one of which is the API docs page link

https://docs.pact.io/pact_broker/advanced_topics/api_docs

however is this the particular page you are referring to?

https://docs.pact.io/pact_broker/publishing_and_retrieving_pacts

I've previously mentioned before it would be nice to have the API documented in Swagger and it was on my list on open source tasks back before I joined Pactflow, so I would be worthwhile soon I am sure

Reply:
Well, in this specific case it's about applying tags, not publishing and retrieving pacts. I think we want this? https://docs.pact.io/pact_broker/tags

Totally agreed on moving to OpenAPI (the fancy new name for Swagger). There's a great tool called Stoplight (https://stoplight.io/) that you might take advantage of, we're starting to use it at eBay